src/nodes/CalloutsContainer.js

```diff
@@ -0,0 +1,109 @@
+/*
+ * @copyright Copyright (c) 2022 Vinicius Reis <vinicius@nextcloud.com>
+ *
+ * @author Vinicius Reis <vinicius@nextcloud.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import { Node, mergeAttributes, isNodeActive } from '@tiptap/core'
+import { typesAvailable } from '../markdownit/containers'
+
+export default Node.create({
+
+	name: 'calloutContainer',
+
+	content: 'paragraph+',
```

Comment:
Just curious, is limiting to paragraph done on purpose or would there be also a way to have other blocks like headings/lists in the callout container as well? Fine to keep it but if there are limitations maybe a quick code comment might be good about the reasoning ;)

Reply:
@vinicius73 and me came up with this during a call.
One of the issues we were facing was having different callouts nested - like a warning inside an info. That would be super confusing.

I also thought it would be good to try and keep the schema as simple as possible to make it easier to reason about. I was afraid we might create such complex structures in text that the generated markdown will not be parsed by other markdown consumers. Today i tried to create such a document by nesting lists and quotes etc. and the resulting markdown was parsed well by github. So markdown is able to handle much more complexity than i thought.

So now i think we actually need to allow for the same level of complexity in the schema. If we have a file that has content that the schema does not allow for it will fail to load in text. In case of the callouts this would currently already be invalid:
```
::: info
### heading in the info
Some text
:::
```

So I think we should allow for all block content in infoboxes to be able to parse markdown that contains these.

At the same time I would like to try to discourage creating too complex structures in the UI. For example... when you are in an info box and click the warning box button... we turn the infobox into a warning box rather than adding a warning box inside the info box.

Reply:
I'd be fine either way, but iirc yesterday (in our call after the message) you mentioned that now you think that supporting different node types would be more fitting to what markdown can actually represent.

We could also get this in first with that limitation and look into the further sub-node types in a follow up.